drv/Collection.drv

```diff
@@ -167,8 +167,7 @@ public interface COLLECTION KEY_GENERIC extends Collection<KEY_GENERIC_CLASS>, K
 	 * @see Collection#removeIf(java.util.function.Predicate)
 	 * @return {@code true} if any elements were removed.
 	 */
-	@SuppressWarnings("overloads")
-	default boolean removeIf(final JDK_PRIMITIVE_PREDICATE filter) {
+	default boolean remIf(final JDK_PRIMITIVE_PREDICATE filter) {
```

Comment:
should be `removeIf` (this change would break public API!)

Reply:
But that fixes the "@SuppressWarnings("overloads")" bug that was present in the first place where casting was required.

Reply:
How is that a bug? You can just write `(int arg) -> arg < 0` instead of `(IntPredicate) arg -> arg < 0` (or something like that) to disambiguate the call btw. Breaking public API is not good.

Reply:
![image](https://user-images.githubusercontent.com/2556269/96931304-5d341180-14bd-11eb-835d-82e5df51de84.png)
Yeah works like a charm.

Reply:
Yes, I know this doesn't work :-) `data.removeIf((int t) -> t == 0);` does.

Reply:
![image](https://user-images.githubusercontent.com/2556269/96936661-c79d7f80-14c6-11eb-9ffa-4cc53621c9a9.png)
Yeah works wonderfully with more optimized lambdas.

Reply:
`data.removeIf((IntPredicate) this::removeIf)` works.

My Point being: Do you have to break the public API in order to save a few characters which your IDE can automatically insert?

Reply:
So you suggest implementing functions that causes compiler warnings is a good idea in general? Because that is what you are suggesting. I mean your solution works, but the point of the change is to remove the casting requirement in general. Else why not change rem function or removeInt to remove because that's exactly the same thing.

Reply:
Honestly @vigna should decide on this one.

Reply:
Its not a compiler warning, but a compiler error. And no, its far from being the same thing.

And yes of course he should decide, on all of these. I'm just trying to minimize his effort by giving you pointers for things which are very likely to cause problems. For example, changing public API for a small patch. A lot of people use this library, you can't just rename API methods without a lot of consideration. Yes it would be good if all of these methods are free of ambiguities but that is not something to be done in a minor patch.

Reply:
Yeah, totally. Changing the public API is a no-no.